src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml

```diff
@@ -314,6 +314,20 @@
 <!--            <tag name="api_platform.filter" />-->
 <!--        </service>-->
 
+        <service id="sylius_api.search_filter.promotion_coupon" parent="api_platform.doctrine.orm.search_filter" public="true">
+            <argument type="collection">
+                <argument key="code">exact</argument>
+            </argument>
+            <tag name="api_platform.filter" />
+        </service>
+
+        <service id="sylius_api.search_filter.promotion_coupon_promotion" parent="api_platform.doctrine.orm.search_filter" public="true">
+            <argument type="collection">
+                <argument key="promotion.code">exact</argument>
+            </argument>
+            <tag name="api_platform.filter" />
+        </service>
```

Comment:
```suggestion
        <service id="sylius_api.search_filter.promotion_coupon" parent="api_platform.doctrine.orm.search_filter" public="true">
            <argument type="collection">
                <argument key="code">exact</argument>
                <argument key="promotion.code">exact</argument>
            </argument>
            <tag name="api_platform.filter" />
        </service>
```

Reply:
It is possible to have a promotion coupon with a code identical to a different promotion, so the filters should be separate.

Reply:
There isn't an ADR file for that yet, but we have decided to create one filter per resource with one or many arguments. This approach makes it easier to change in the end application.

Reply:
Is it only for the "search" filter?

Reply:
It includes all API platform filters such as `search_filter`, `order_filter`, `boolean_filter`, `date_filter` itp.